.circleci/config.yml

```
@@ -26,11 +28,13 @@ jobs:
       - checkout
       - restore_cache:
           keys:
-            - composer-v1-{{ checksum "composer.lock" }}
+            - composer-v1-{{ .Revision }}
```

Could you try using the filename instead of the revision? This works fine for jobs triggered by the same commit, but I don't think it will be really performant between PRs for example, as the revision will constantly change.

```suggestion
            - composer-v1-{{ checksum "composer.json" }}
```

---

I changed this. In theory, we could run into a situation in which the restored `vendor` directory has an old version of a dependency (up to 15 days old). Since we run `composer install`, I believe it would not update if the currently installed version met the version constraint in `composer.json`. Admittedly, this is an edge case and would only occur if a dependency issued a patch release to fix a bug that is impacting our tests. In that scenario, we could update the `composer.json` to indicate the minimum required patch level, though.

---

I might be wrong, but, does not `composer install` behave like `composer update` if no `composer.lock` file is present? `install` needs to resolve the dependencies first and then will install or update depending on the version prevent in `vendor` folder.

---

I wondered that myself, so I deleted my local `composer.json` but left the vendor directory (simulating what should happen in the job after the cached vendor directory is restored). Running `composer install` resulted in `Nothing to install or update`. I did not try having an intentionally outdated vendor directory and then running install, however.

---

Indeed it install nothing, but first Composer needs to know what to install to be able to compare with the content of the `vendor` folder. See the difference with [composer.lock](https://app.circleci.com/pipelines/github/arkaitzgarro/elastic-apm-laravel/310/workflows/bdaf9b2c-52c5-43e6-b438-932c89d2a0cc/jobs/852) file and [without](https://app.circleci.com/pipelines/github/arkaitzgarro/elastic-apm-laravel/324/workflows/8ab55e97-c669-4975-995b-5300425be17f/jobs/899). Same console output, completely different execution times :)